internal/services/kusto/kusto_eventgrid_data_connection_resource.go

```diff
@@ -137,6 +137,28 @@ func resourceKustoEventGridDataConnection() *pluginsdk.Resource {
 					string(kusto.EventGridDataFormatW3CLOGFILE),
 				}, false),
 			},
+
+			"database_routing": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  string(kusto.DatabaseRoutingSingle),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(kusto.DatabaseRoutingSingle),
+					string(kusto.DatabaseRoutingMulti),
+				}, false),
+			},
+
+			"eventgrid_resource_id": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+			},
+
+			"managed_identity_resource_id": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
Can we improve the validation here? Here's an example: https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/kusto/kusto_eventhub_data_connection_resource.go#L116

Reply:
Done